### PR TITLE
GoQuorum ATs: run the estimate gas test

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

Revert #2282 
Enabling one more passing GoQuorum AT 
See https://github.com/hyperledger/besu/pull/2282

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java [hunk 1]
```diff
@@ -163,25 +157,12 @@ public TransactionSimulator(
         callParams.getGasLimit() >= 0 ? callParams.getGasLimit() : header.getGasLimit();
     final Wei gasPrice = callParams.getGasPrice() != null ? callParams.getGasPrice() : Wei.ZERO;
     final Wei value = callParams.getValue() != null ? callParams.getValue() : Wei.ZERO;
-
-    // If GoQuorum privacy enabled, and value = zero, do simulation with max non-zero bytes.
-    // It is possible to have a data field that has a lower intrinsic value than the PTM hash
-    // so this checks the tx as if we were to place a PTM hash (with all non-zero values).
-    // This means a potential over-estimate of gas, rather than the exact cost to run right now.
-    final Bytes payload =
```
Comment (mark-terry) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java:171]: This isn't needed now?

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java [hunk 2]
```diff
@@ -198,6 +179,10 @@ public TransactionSimulator(
 
     final Transaction transaction = transactionBuilder.build();
 
+    final ProtocolSpec protocolSpec = protocolSchedule.getByBlockNumber(header.getNumber());
+
+    final MainnetTransactionProcessor transactionProcessor =
```
Comment (mark-terry) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java:184]: What's the impact of moving this from before, to after the transaction is built?

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java (hunk 1 above)
Comment (macfarla) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java:171]: yeah this PR is reverting my previous one

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java (hunk 2 above)
Comment (macfarla) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java:184]: no impact. I can move it back

**Review by mark-terry — APPROVED** [review]
LGTM.

**Pull request merged**